src/binder_sms.c

```diff
@@ -388,47 +419,93 @@ binder_sms_submit_cb(
          * sendSMSExpectMoreResponse(RadioResponseInfo, SendSmsResult sms);
          * sendImsSmsResponse(RadioResponseInfo, SendSmsResult sms);
          */
-        if (resp == RADIO_RESP_SEND_SMS ||
-            resp == RADIO_RESP_SEND_SMS_EXPECT_MORE ||
-            resp == RADIO_RESP_SEND_IMS_SMS) {
-            if (error == RADIO_ERROR_NONE) {
-                const RadioSendSmsResult* res;
-                GBinderReader reader;
-
-                gbinder_reader_copy(&reader, args);
-                res = gbinder_reader_read_hidl_struct(&reader,
-                    RadioSendSmsResult);
+        if (self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+            if (resp == RADIO_RESP_SEND_SMS ||
+                resp == RADIO_RESP_SEND_SMS_EXPECT_MORE ||
+                resp == RADIO_RESP_SEND_IMS_SMS) {
+                if (error == RADIO_ERROR_NONE) {
+                    const RadioSendSmsResult* res;
+                    GBinderReader reader;
+
+                    gbinder_reader_copy(&reader, args);
+                    res = gbinder_reader_read_hidl_struct(&reader,
+                        RadioSendSmsResult);
+
+                    if (res) {
+                        DBG("%ssms msg ref: %d, ack: %s err: %d", ims ? "ims " : "",
+                            res->messageRef, res->ackPDU.data.str, res->errorCode);
+
+                        /*
+                         * Error is -1 if unknown or not applicable,
+                         * otherwise 3GPP 27.005, 3.2.5
+                         */
+                        if (res->errorCode > 0) {
+                            err.type = OFONO_ERROR_TYPE_CMS;
+                            err.error = res->errorCode;
+                        } else {
+                            /* Success */
+                            cb(binder_error_ok(&err), res->messageRef, cbd->data);
```

Comment:
The `cb()` exists always in every case and return value is `void`/uninteresting?

Reply:
Cb is set by ofono and is always valid.